File: sep-aggregator-service/src/main/java/com/kurtcan/sepaggregatorservice/shared/entity/BaseEntity.java
package com.kurtcan.sepaggregatorservice.shared.entity;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.SuperBuilder;

import java.time.Instant;
import java.util.UUID;

@Data
@SuperBuilder
@NoArgsConstructor
@AllArgsConstructor
public class BaseEntity implements DbEntity {
    private UUID id;
    private Instant createdAt;
    private Instant updatedAt;
}

File: sep-search-service/src/main/kotlin/com/kurtcan/sepsearchservice/base/BaseEntity.kt
package com.kurtcan.sepsearchservice.base

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonProperty
import com.kurtcan.sepsearchservice.shared.elasticsearch.ElasticsearchEntity
import com.kurtcan.sepsearchservice.shared.entity.DbEntity
import java.time.Instant
import java.util.*

open class BaseEntity @JsonCreator constructor(
    @JsonProperty("id") private var id: UUID? = null,
    @JsonProperty("createdAt") private var createdAt: Instant? = null,
    @JsonProperty("updatedAt") private var updatedAt: Instant? = null
) : ElasticsearchEntity, DbEntity {
    override fun getId(): UUID {
        return id!!
    }

    fun setId(newId: UUID) {
        id = newId
    }

    override fun getCreatedAt(): Instant {
        return createdAt!!
    }

    fun setCreatedAt(newCreatedAt: Instant) {
        createdAt = newCreatedAt
    }

    override fun getUpdatedAt(): Instant {
        return updatedAt!!
    }

    fun setUpdatedAt(newUpdatedAt: Instant) {
        updatedAt = newUpdatedAt
    }
}